### PR TITLE
Bump otel version and get ready for stable metrics release.

### PR DESCRIPTION
**Pull request by jsuereth**
- BREAKING: metrics now written to `workload.googleapis.com`
- BREAKING: metrics now only supports `endpoint` not `stub` config.
- FEATURE: trace now supports `stub` config.
- Clean up dependencies of metric exporter to bare minimum / avoid unstable dependencies (semconv)
- Clean up javadoc for metrics exporter

### CODE REVIEW

**Comment by dgnemo** [comment]
Any thoughts on renaming `com.google.cloud.opentelemetry.metric.MetricExporter` to 
`com.google.cloud.opentelemetry.metric.GoogleCloudMetricExporter`  before releasing the stable release?

that would prevent the import name "conflict" with `io.opentelemetry.sdk.metrics.export.MetricExporter` that is forcing the user to specify the whole package in the code?
(see https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/tree/main/exporters/metrics#readme )

File: exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricConfiguration.java [hunk 1]
```diff
@@ -96,16 +120,24 @@ public abstract static class Builder {
 
     abstract Duration getDeadline();
 
+    /** Set the GCP project where metrics should be writtten. */
```
Comment (dashpole) [inline exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricConfiguration.java:122]:
```suggestion
    /** Set the GCP project where metrics should be writtten. Defaults to the application default credential's project.*/
```
or something like that.

File: exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricConfiguration.java [hunk 2]
```diff
@@ -96,16 +120,24 @@ public abstract static class Builder {
...
     public abstract Builder setProjectId(String projectId);
 
+    /** Set the credentials to use when writing metrics. */
     public abstract Builder setCredentials(Credentials newCredentials);
 
-    public abstract Builder setMetricServiceStub(MetricServiceStub newMetricServiceStub);
-
+    /** Set the deadline for exporting batches of metric timeseries. */
```
Comment (dashpole) [inline exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricConfiguration.java:128]: Defaults to 12s?

File: exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java [hunk 3]
```diff
@@ -37,54 +37,24 @@
...
 import org.slf4j.LoggerFactory;
 
-public class MetricTranslator {
+/** Utility methods to translate metrics from OTEL to GCM format. */
+public final class MetricTranslator {
 
   private static final Logger logger = LoggerFactory.getLogger(MetricTranslator.class);
 
-  static final String DESCRIPTOR_TYPE_URL = "custom.googleapis.com/OpenTelemetry/";
+  static final String DESCRIPTOR_TYPE_URL = "workload.googleapis.com/";
```
Comment (dashpole) [inline exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java:51]: is this configurable, or should it be?  We can tackle that at a later point if needed.

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java [hunk 4]
```diff
@@ -53,27 +51,39 @@ public class EndToEndTest {
...
+    exporter =
+        MetricExporter.createWithConfiguration(
+            MetricConfiguration.builder()
+                .setMetricServiceEndpoint(mockContainer.getMetricServiceEndpoint())
+                .setInsecureEndpoint(true)
+                .setDescriptorStrategy(MetricDescriptorStrategy.ALWAYS_SEND)
+                .setProjectId(aProjectId)
+                .build());
+
+    // exporter = new MetricExporter(aProjectId, mockClient, MetricDescriptorStrategy.ALWAYS_SEND);
```
Comment (dashpole) [inline exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java:72]: nit: remove commented out code

**Review by dashpole — APPROVED** [review]
lgtm overall

File: exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java (hunk 3 above)
Comment (jsuereth) [reply exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java:51]: It is not.  We can tackle later I think, as it wasn't configurable before.

**Comment by jsuereth** [comment]
@dgnemo This is a good suggestion.  I'll add that before submitting.

**Pull request merged**